### PR TITLE
Feature 62 drag and drop component

### LINKED ISSUE #62: Editor: Drag & drop to add new fields and components
As a user, I'd like to be able to add new fields and components to Metaform with drag & drop action. I would also like to change the order of fields and sections on Metaform using the same technique.

### PR DESCRIPTION
**Pull request by Chi-Chu319**
resolves #62

### CODE REVIEW

File: src/styles/generics/drag-handle/section-hrag-handle.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { createStyles } from "@material-ui/core";
```
Comment (EVP1989) [inline src/styles/generics/drag-handle/section-hrag-handle.ts:1]: Fix typo in file name

File: src/utils/metaform.ts [hunk 2]
```diff
@@ -29,4 +29,30 @@ export default class MetaformUtils {
...
   }
 
+  /**
+   * Create a default field with field Type
+   * 
+   * @param fieldType metaform field type
+   * @returns created field type
+   */
+  public static metaformDefaultField = (fieldType: MetaformFieldType): MetaformField => {
+    // TODO radio, select
```
Comment (EVP1989) [inline src/utils/metaform.ts:39]: Todos should be resolved before merge

File: src/utils/metaform.ts [hunk 3]
```diff
@@ -29,4 +29,30 @@ export default class MetaformUtils {
...
+    } 
+  }
+
+  /**
+   * Convert a empty section
+   * 
+   * @returns created section 
+   */
+  public static metaformDefaultSection = (): MetaformSection => {
+    // TODO radio, select
```
Comment (EVP1989) [inline src/utils/metaform.ts:52]: Todos should be resolved before merge

File: src/localization/fi.json [hunk 4]
```diff
@@ -137,9 +135,23 @@
...
+      "html": "html",
+      "text": "text",
+      "radio": "radio",
+      "select": "select",
+      "number": "number",
+      "memo": "memo",
+      "checkList": "checkList",
+      "submit": "submit",
+      "dateTime": "dateTime",
+      "autoComplete": "autoComplete"
```
Comment (EVP1989) [inline src/localization/fi.json:153]: Once the visual display components are ready, strings should be localized correctly

File: src/components/screens/form-edit-screen.tsx [hunk 5]
```diff
@@ -327,95 +467,345 @@ export class FormEditScreen extends React.Component<Props, State> {
...
     );
   }
 
+  /**
+   * Event handler for drag start
+   * 
+   * @param initial drag start data
+   * @param provided responder provided
+   */
+  private onDragStart = (initial: DragStart, provided: ResponderProvided) => {
```
Comment (EVP1989) [inline src/components/screens/form-edit-screen.tsx:550]: Form edit screen component is getting pretty big. Maybe should consider splitting the code. DnD methods could be located on separate file for example.

**Review by EVP1989 — CHANGES_REQUESTED** [review]

File: src/components/generic/editable-field-components/radio-field-component.tsx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+    return (
+      <Radio
+        color="primary"
+        key={ `${ fieldId }-${ option.name }-radio` }
+        id={ `${ fieldId }-${ option.name }` } 
+        aria-labelledby={ fieldLabelId }
+        name={ field!.name }
+        title={ field!.title }
+        required={ field!.required }
```
Comment (anttileppa) [inline src/components/generic/editable-field-components/radio-field-component.tsx:102]: Field is not optional

```suggestion
        name={ field.name }
        title={ field.title }
        required={ field.required }
```

File: src/components/screens/form-edit-screen.tsx [hunk 7]
```diff
@@ -38,16 +38,24 @@ interface Props extends WithStyles<typeof styles> {
...
+  draggingSection: boolean;
+  selectedFieldIndex?: number;
+  selectedSectionIndex?: number;
 }
 
 /**
  * Component for editing Metaform
  */
 export class FormEditScreen extends React.Component<Props, State> {
+  editorRef: React.RefObject<HTMLDivElement>;
```
Comment (anttileppa) [inline src/components/screens/form-edit-screen.tsx:57]:
```suggestion
  private editorRef: React.RefObject<HTMLDivElement>;
```

File: src/localization/fi.json [hunk 8]
```diff
@@ -143,11 +141,22 @@
...
+    "html": "html",
+    "text": "text",
+    "radio": "radio",
+    "select": "select",
+    "number": "number",
+    "memo": "memo",
+    "checkList": "checkList",
+    "submit": "submit",
+    "dateTime": "dateTime",
+    "autocomplete": "autocomplete"
```
Comment (anttileppa) [inline src/localization/fi.json:160]: Should these be in finnish?

**Review by anttileppa — APPROVED** [review]

File: src/localization/fi.json (hunk 8 above)
Comment (Chi-Chu319) [reply src/localization/fi.json:160]: Discussed with Veli-matti before the name of form element could be in English

**Pull request merged**